### ISSUE TITLE
Fails to compile in M1

### ISSUE THREAD

**Issue by adria0** [iden3/circom#17]
It can be temporally fixed by compling, keeping the arm64 code from the libwabt.a fat binary and compile again

```
# for commit a9a4a24e4e95debb0230a233c259fac7d3e1e186
cargo build --release
lipo target/release/build/wabt-sys-ccf4f394b6bb7490/out/build/libwabt.a -thin arm64 -output target/release/build/wabt-sys-ccf4f394b6bb7490/out/build/libwabt.a
cargo build --release
```

**Comment by alrubio** [comment]
We have installed circom in M1 without any problem. I don't know why you have problems with that. As it is related to wabt it might be due to your version of cmake. Which is the compilation error you get?

**Comment by adria0** [comment]
The error is

```
   Compiling wabt v0.10.0
error: failed to add native library /Users/adriamassanet/w/amb/circom/target/release/build/wabt-sys-17454c87130c2294/out/build/libwabt.a: file too small to be an archive

error: could not compile `wabt-sys` due to previous error
warning: build failed, waiting for other jobs to finish...
```

my rust & cmake says

```
➜  circom git:(master) cargo version
cargo 1.56.0 (4ed5d137b 2021-10-04)
➜  circom git:(master) cmake --version
cmake version 3.20.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
➜  circom git:(master) type cmake
cmake is /opt/homebrew/bin/cmake
```

**Comment by miguelis** [comment]
I am also using a M1 chip and I didn't need to do anything out of the ordinary to compile correctly circom. My cargo version is the same as yours and the cmake version is 3.21.3. I also didn't need to use lipo because using lipo -archs ..../libwabt.a shows that it is already generated for arm64 architecture. Perhaps you already had it installed for x86 for other projects. 

However, I am also having a problem in M1 during the witness generation from the C++ files. Do you have similar problems compiling the C++ files to generate the witness?

**Comment by adria0** [comment]
Fixed for the latest versions `cmake 3.22.0` && `rustc 1.56.1` (`brew install cmake` && `rustup update`).

I close the issue.

@miguelis, I did not tried to compile the witness with C++ yet in M1
